src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPropertySymbol.cs

```diff
@@ -7,338 +7,143 @@
-                DiagnosticBag diagnostics)
-            {
-                _property = property;
-                Name = SourcePropertyAccessorSymbol.GetAccessorName(
-                    paramName,
-                    getNotSet: false,
-                    // https://github.com/dotnet/roslyn/issues/44684
-                    isWinMdOutput: false);
-
-                var comp = property.DeclaringCompilation;
-                var type = TypeWithAnnotations.Create(comp.GetSpecialType(SpecialType.System_Void));
-                var initOnlyType = Binder.GetWellKnownType(
-                    comp,
-                    WellKnownType.System_Runtime_CompilerServices_IsExternalInit,
-                    diagnostics,
-                    property.Location);
-                var modifiers = ImmutableArray.Create<CustomModifier>(CSharpCustomModifier.CreateRequired(initOnlyType));
-
-                ReturnTypeWithAnnotations = type.WithModifiers(modifiers);
-            }
-
-            internal override bool IsInitOnly => true;
-
-            public override MethodKind MethodKind => MethodKind.PropertySet;
-
-            public override int Arity => 0;
-
-            public override bool IsExtensionMethod => false;
-
-            public override bool HidesBaseMethodsByName => false;
-
-            public override bool IsVararg => false;
-
-            public override bool ReturnsVoid => true;
-
-            public override bool IsAsync => false;
-
-            public override RefKind RefKind => RefKind.None;
-
-            public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
-
-            public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations => ImmutableArray<TypeWithAnnotations>.Empty;
+            // Nothing to do here
+        }
 
-            public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+        protected override DeclarationModifiers MakeModifiers(
+            SyntaxTokenList modifiers,
+            bool isExplicitInterfaceImplementation,
+            bool isIndexer,
+            bool accessorsHaveImplementation,
+            Location location,
+            DiagnosticBag diagnostics,
+            out bool modifierErrors)
+        {
+            Debug.Assert(!isExplicitInterfaceImplementation);
+            Debug.Assert(!isIndexer);
+            var defaultAccess = DeclarationModifiers.Public;
+
+            // No modifiers are allowed on synthesized property symbols
+            var allowedModifiers = DeclarationModifiers.None;
+
+            var mods = ModifierUtils.MakeAndCheckNontypeMemberModifiers(
+                modifiers,
+                defaultAccess,
+                allowedModifiers,
+                location,
+                diagnostics,
+                out modifierErrors);
+
+            Debug.Assert(!mods.HasFlag(DeclarationModifiers.Unsafe));
+            return mods;
```

Comment:
>return mods; [](start = 12, length = 12)

I think the result should be hardcoded. We know what DeclarationModifiers should be. There is no reason to call all that helpers and it is hard to say if something is missing.